### DISCUSSION TITLE
RuntimeError: dictionary changed size during iteration

### DISCUSSION THREAD

**Discussion by wd15** [usnistgov/fipy#852] [General]
[This error](https://hydra.nixos.org/build/172501646/nixlog/2) started appearing recently on Nix's test site.

```
======================================================================
ERROR: import fipy.meshes.gmshMesh
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/build/FiPy-3.4.2.1/fipy/tests/lateImportTest.py", line 26, in setUp
    self.suite.addTest(self._getTestSuite(module = module))
  File "/build/FiPy-3.4.2.1/fipy/tests/doctestPlus.py", line 144, in _getTestSuite
    return doctest.DocTestSuite(module,
  File "/nix/store/9px00aaqzb6n5p03i9wd8rx3msg95y9r-python3-3.9.11/lib/python3.9/doctest.py", line 2378, in DocTestSuite
    tests = test_finder.find(module, globs=globs, extraglobs=extraglobs)
  File "/nix/store/9px00aaqzb6n5p03i9wd8rx3msg95y9r-python3-3.9.11/lib/python3.9/doctest.py", line 939, in find
    self._find(tests, obj, name, module, source_lines, globs, {})
  File "/nix/store/9px00aaqzb6n5p03i9wd8rx3msg95y9r-python3-3.9.11/lib/python3.9/doctest.py", line 995, in _find
    for valname, val in obj.__dict__.items():
RuntimeError: dictionary changed size during iteration

----------------------------------------------------------------------
```

I'll disable this test on Nix for now until I figure it out.

Also, getting this at the end of the tests

```
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
Skipped 28 doctest examples because `gmsh` cannot be found on the $PATH
Skipped 83 doctest examples because the `tvtk` package cannot be imported
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

indicating that gmsh isn't being found. Versions are as follows:

```
fipy version 3.4.2.1
numpy version 1.21.5
pysparse is not installed
scipy version 1.8.0
matplotlib version 3.5.1
mpi4py version 3.1.3
petsc4py is not installed
pyamgx is not installed
PyTrilinos is not installed
enthought.mayavi is not installed
gmsh is not installed
```

Gmsh is installed, but is at 4.9.2.

@guyer, would some of this be fixed with your latest changes?

**Comment by guyer** [comment]
This was fixed by #848

**Reply by wd15** [reply]
Great. Is a release on the cards? I could also patch it in Nix. Would it just require patching the changes in [gmshMesh.py](https://github.com/usnistgov/fipy/pull/848/files#diff-3e475d9f0271bc65e1e1343938db53a04442f91d9d1786213f2ac513d29f3f9e) or more complicated?

**Reply by wd15** [reply]
Also, what's the best way to switch off a test? Is it possible from the command line?

**Comment by guyer** [comment]
Yeah, a release is probably due. There are two more CI failures that should be addressed, first (see #849 & #850).

The fix was a little bit more complicated to be robust across versions, but it could be achieved by patching gmshMesh.py.

**Comment by guyer** [comment]
Nothing that I’m aware of. Probably a good reason to migrate to nose or pytest or whatever it is that we’re supposed to be using this week.

**Comment by wd15** [comment]
Just to wind this up. [Now repaired in nixpkgs with a patch](https://github.com/NixOS/nixpkgs/pull/167935).
